api/app/Observers/PoolObserver.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Observers;
+
+use App\Enums\PoolStatus;
+use App\Models\Pool;
+use App\Notifications\ApplicationDeadlineExtended;
+use Illuminate\Support\Facades\Log;
+use Throwable;
+
+class PoolObserver
+{
+    /**
+     * Handle the Pool "created" event.
+     */
+    public function created(Pool $pool): void
+    {
+        //
+    }
+
+    /**
+     * Handle the Pool "updated" event.
+     */
+    public function updated(Pool $pool): void
+    {
+        // Notify users with a draft application when published pool's closing date is extended
+        $oldClosingDate = $pool->getOriginal('closing_date');
+        $newClosingDate = $pool->closing_date;
+
+        // Check if published pool's new closing date is greater than old date
+        if (
+            $pool->status === PoolStatus::PUBLISHED->name
+            && ! is_null($oldClosingDate)
+            && ! is_null($newClosingDate)
+            && $newClosingDate->gte($oldClosingDate)
+        ) {
+
+            $pool = Pool::find($pool->id);
+            $pool->poolCandidates
+                ->where('pool_candidate_status', 'DRAFT') // Only send notification to draft applications
+                ->each(function ($poolCandidate) use ($newClosingDate) {
+
+                    $notification = new ApplicationDeadlineExtended(
+                        $poolCandidate->user->getFullName(),
```

Comment:
I got a lazy load error.  Probably from this line.
`Attempted to lazy load [user] on model [App\Models\PoolCandidate] but lazy loading is disabled.`